src/fhirtypes/ElementDefinition.ts

```diff
@@ -1387,15 +1396,24 @@ export class ElementDefinition {
       if (newElements.length === 0) {
         // If it has a profile, use that, otherwise use the code
         const type = this.type[0].profile?.[0] ?? this.type[0].code;
-        const def = resolve(type);
-        newElements = def?.elements.slice(1).map(e => {
-          const eClone = e.clone();
-          eClone.id = eClone.id.replace(def.type, `${this.id}`);
-          eClone.structDef = this.structDef;
-          // Capture the original so that diffs only show what changed *after* unfolding
-          eClone.captureOriginal();
-          return eClone;
-        });
+        const json = fisher.fishForFHIR(
```

Comment:
I think this is right, because based on the line above it, `type` could be referring to `this.type[0].code` or `this.type[0].profile`.  If it is `this.type[0].code` then it is most-likely a Type (e.g., `Quantity`) but _could_ be a Resource (there are a few rare cases of inline resources).  If it is `this.type[0].profile`, it could be a Profile (on a Type or Resource) or a specific Extension (which is more or less a profile on `Extension`).  So I think we need to support all four of those types.  I guess we just don't have tests that use inline resources or profiles in this spot.